public/app/features/alerting/unified/components/rules/RuleStateHistory.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { FC } from 'react';
+
+import { dateTimeFormat, GrafanaTheme } from '@grafana/data';
+import { css } from '@emotion/css';
+import { StateHistoryItem } from 'app/types/unified-alerting';
+import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '../DynamicTable';
+import { AlertStateTag } from './AlertStateTag';
+import { useManagedAlertStateHistory } from '../../hooks/useManagedAlertStateHistory';
+import { LoadingPlaceholder, useStyles } from '@grafana/ui';
+import { LegacyAlertStateToGrafanaAlertState } from '../../utils/rules';
+import { AlertLabel } from '../AlertLabel';
+
+interface RuleStateHistoryProps {
+  alertId: string;
+}
+
+type PartialHistoryItem = Pick<StateHistoryItem, 'newState' | 'updated' | 'data'>;
+
+const RuleStateHistory: FC<RuleStateHistoryProps> = ({ alertId }) => {
+  const { loading, error, result = [] } = useManagedAlertStateHistory(alertId);
+
+  if (loading && !error) {
+    return <LoadingPlaceholder text={'Loading history...'} />;
+  }
+
+  if (error && !loading) {
+    return <div>{error.toString()}</div>;
+  }
+
+  const columns: Array<DynamicTableColumnProps<PartialHistoryItem>> = [
+    {
+      id: 'state',
+      label: 'State',
+      renderCell: renderStateCell,
+    },
+    {
+      id: 'value',
+      label: '',
+      renderCell: renderValueCell,
+    },
+    { id: 'timestamp', label: 'Time', renderCell: renderTimestampCell },
+  ];
+
+  const items: Array<DynamicTableItemProps<PartialHistoryItem>> = result.map((historyItem) => ({
+    id: historyItem.id,
+    data: {
+      newState: historyItem.newState,
```

Comment:
Maybe I should just pass the entire `historyItem` and not bother with `PartialHistoryItem`? Thoughts?